File: 75_sort_colors/sortColors.ts
/**
 Do not return anything, modify nums in-place instead.
 */
function sortColors(nums: number[]): void {
  const count = new Array<number>(4).fill(0);

  for (let num of nums) {
    count[num]++;
  }

  let pos = 0;

  for (let i = 0; i < count.length; i++) {
    for (let j = 0; j < count[i]; j++) {
      nums[pos] = i;
      pos++;
    }
  }
}

File: 75_sort_colors/sortColors.go
func sortColors(nums []int) {
	count := [3]int{}

	for _, num := range nums {
		count[num]++
	}

	k := 0

	for i := range 3 {
		for j := 0; j < count[i]; j++ {
			nums[k] = i
			k++
		}
	}
}